ILTransform/TestProject.cs

```diff
@@ -2202,16 +2211,50 @@ string InterestingNamespace(TestProject project)
             int dirSize = (dirAttempt != null) && dirAttempt.AllUnique() ? dirAttempt.Select(s => s.Length).Sum() : int.MaxValue;
 
             int bestSize = Math.Min(filenameSize, Math.Min(projectSize, dirSize));
+            List<string> bestAttempt = new List<string>();
+            
             if (bestSize == int.MaxValue)
             {
-                Console.WriteLine("No simple namespace renames for projects");
-                return;
+                // Try see whether we can merge the results or use some index  
+                var dedupNames = new HashSet<string>();
+                var filenameCounter = new Dictionary<string, int>();
+                foreach ((string filename, string projectName) in filenameAttempt.Zip(projectAttempt))
+                {
+                    if (!string.IsNullOrEmpty(filename) && dedupNames.Add(filename))
+                    {
+                        bestAttempt.Add(filename);
+                    }
+                    else
+                    {
+                        if (dedupNames.Add(projectName))
+                        {
+                            bestAttempt.Add(projectName);
+                        }
+                        else
+                        {
+                            string indexedName = string.IsNullOrEmpty(filename) ? projectName : filename;
+                            if (!filenameCounter.ContainsKey(indexedName))
+                            {
+                                filenameCounter[indexedName] = 1;
+                            }
+                            while (!dedupNames.Add(indexedName + filenameCounter[indexedName]++) && filenameCounter[indexedName] < 1500) { }
+                            bestAttempt.Add(indexedName + (filenameCounter[indexedName] - 1));
```

Comment:
Why would the next index be in the names already?  Is this for handling something like (A\foo.ilproj, B\foo.ilproj, C\foo.ilproj, D\foo2.ilproj)?  What is special about 1500?  Do you need an error check if it gets to 1500, or are you replying on it eventually failing the uniqueness check below?

I think this could be simplified - something like
```
                            int counter = filenameCounter.GetValueOrDefault(indexedName, 1);
                            while (!dedupNames.Add(indexedName + counter) && filenameCounter[indexedName] < 1500) { counter++; }
                            filenameCounter[indexedName] = counter;
                            bestAttempt.Add(indexedName + counter);

Reply:
It is just in case there are indexes already on proj files. The 1500 shouldn't be reached, we can remove it, I added it as a way of ensuring termination as well as getting notified (finding duplicated cases) if we reach that point.